### ISSUE TITLE
Error report on Marketplace

### ISSUE THREAD

**Issue by AObuchow** [AObuchow/Eclipse-Spectrum-Theme#104]
There's some[ error reports I've been seeing on marketplace](https://marketplace.eclipse.org/content/error/report/5158108):
```
Listings: 
Spectrum Dark Theme
Error Code: 
4
Error Message: 
Operation details
Installable Units Reported: 
com.aobuchow.themes.spectrum.feature.feature.group,1.0.0.202006290354
Detailed Message: 
Cannot complete the install because one or more required items could not be found. Software being installed: Spectrum Theme 1.0.0.202006290354 (com.aobuchow.themes.spectrum.feature.feature.group 1.0.0.202006290354) Missing requirement: Spectrum Theme 1.0.0.202006290354 (com.aobuchow.themes.spectrum 1.0.0.202006290354) requires 'bundle org.eclipse.e4.ui.css.swt.theme 0.0.0' but it could not be found Cannot satisfy dependency: From: Spectrum Theme 1.0.0.202006290354 (com.aobuchow.themes.spectrum.feature.feature.group 1.0.0.202006290354) To: com.aobuchow.themes.spectrum [1.0.0.202006290354]
```
It might be related to not specifying a version for org.eclipse.e4.ui.css.swt.theme? See: https://github.com/AObuchow/Eclipse-Spectrum-Theme/blob/master/com.aobuchow.themes.spectrum/META-INF/MANIFEST.MF#L9

**Comment by ingomohr** [comment]
@AObuchow It could not hurt to give it a min version. Although, not specifying any version typically allows any version. But, maybe the Eclipse version the user has doesn't have access to any version of ``org.eclipse.e4.ui.css.swt.theme``.

**Comment by AObuchow** [comment]
@mbooth101 any ideas? :)

**Comment by AObuchow** [comment]
> @AObuchow It could not hurt to give it a min version. Although, not specifying any version typically allows any version. But, maybe the Eclipse version the user has doesn't have access to any version of `org.eclipse.e4.ui.css.swt.theme`.

I'll try setting a min version and see if it helps. I wonder why someone wouldn't have `org.eclipse.e4.ui.css.swt.theme` as it's part of platform?

**Comment by ingomohr** [comment]
> > @AObuchow It could not hurt to give it a min version. Although, not specifying any version typically allows any version. But, maybe the Eclipse version the user has doesn't have access to any version of `org.eclipse.e4.ui.css.swt.theme`.
> 
> I'll try setting a min version and see if it helps. I wonder why someone wouldn't have `org.eclipse.e4.ui.css.swt.theme` as it's part of platform?

Yes, me too. But your plug-in doesn't require any min version, so in theory it should be possible to launch the installer on something like Eclipse Juno, as well (Eclipse 3.8). Or does the Marketplace entry as such restrict access to your plug-in to 2020-06 in a technical way? (I have no experience w/ setting up Marketplace entries).

**Comment by mbooth101** [comment]
> @mbooth101 any ideas? :)

I don't know. It's pretty annoying it doesn't give you details about which version of Eclipse they are using.

**Comment by AObuchow** [comment]
@mbooth101 right! Maybe I'll file a bug regarding adding more info to the error details.

**Comment by mbooth101** [comment]
> @mbooth101 right! Maybe I'll file a bug regarding adding more info to the error details.

In the meantime, I would just make it obvious from the marketplace listing that this github project is the best place to report problems. This anonymous market place error is pretty inactionable IMO, but if users actually come here to report problems themselves, then we can chat to them to extract all the detail necessary to help them fix their problem.

**Comment by AObuchow** [comment]
@mbooth101 +1 that's a good idea, I went ahead and modified the marketplace listing:
![image](https://user-images.githubusercontent.com/10300119/86175743-de702e80-baf1-11ea-8815-d8ce633d952b.png)

I could have maybe worded it better - anyone is free to suggest how it should be written :P

**Comment by AObuchow** [comment]
Unfortunately, I don't think adding a minimum version requirement to org.eclipse.e4.ui.css.swt.theme[ made much of a difference](https://marketplace.eclipse.org/content/error/report/5162336):
```
Cannot complete the install because one or more required items could not be found. Software being installed: Spectrum Theme 1.0.0.202007021914 (com.aobuchow.themes.spectrum.feature.feature.group 1.0.0.202007021914) Missing requirement: Spectrum Theme 1.0.0.202007021914 (com.aobuchow.themes.spectrum 1.0.0.202007021914) requires 'bundle org.eclipse.e4.ui.css.swt.theme 0.12.700' but it could not be found Cannot satisfy dependency: From: Spectrum Theme 1.0.0.202007021914 (com.aobuchow.themes.spectrum.feature.feature.group 1.0.0.202007021914) To: com.aobuchow.themes.spectrum [1.0.0.202007021914]
```

I could try lowering the version requirement, as this might have made only Eclipse 2020-06/4.16 users able to use it... However, _ideally_ they should just upgrade to Eclipse 4.16 🤔

**Comment by mbooth101** [comment]
> However, _ideally_ they should just upgrade to Eclipse 4.16 thinking

You'd be surprised how many people come into the IRC channel under the misapprehension that they are using the latest Eclipse, but it turns out to be Oxygen (4.7) or similar vintage... :-o

**Comment by AObuchow** [comment]
@mbooth101 oh my hahah 😅 Well, if people want the plugin to be compatible with an earlier version of Eclipse they can make a bug I suppose. 

IIRC I tried my theme on Eclipse 4.13 and it had some issues with the tabs. The theme is actually using swt-active-tab-highlight which only came out in 4.16.

**Comment by PyvesB** [comment]
For your information, I'm getting very similar errors with my Planet Themes, though it's for bundle `org.eclipse.ui.themes`, not `org.eclipse.e4.ui.css.swt.theme`.

**Comment by AObuchow** [comment]
@PyvesB thanks for info, good to know :)
